src/App.js

```diff
@@ -20,8 +22,84 @@ import Login from './components/Login';
 import Logout from './components/Logout';
 import Profile from './components/Profile';
 import StyleGuide from './components/StyleGuide/index';
+import Box from './components/Box';
 
 import { styles } from './App.styles';
+import { Emails } from 'constants/Emails';
+
+const ProtectedRoute = ({ backend, children, path }) => {
+  const [verified, setVerified] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [badDomain, setBadDomain] = useState(false);
+
+  const checkVerification = useCallback(() => {
+    if (!backend.isLoggedIn()) {
+      if (!backend.authLoaded) {
+        window.setTimeout(checkVerification, 100);
+        return;
+      }
+    }
+    backend.isValidHealthcareWorker().then((verified) => {
+      if (verified) {
+        console.log('verified');
+        setLoading(false);
+        setVerified(true);
+      } else {
+        setLoading(false);
+        setVerified(false);
+        setBadDomain(backend.badDomain);
+      }
+    });
+  }, [backend]);
+
+  useEffect(() => {
+    checkVerification();
+  }, [checkVerification]);
+
+  let content = <Route path={path}>{children}</Route>;
+  if (loading) {
+    content = (
+      <Box>
+        <div className="alert alert-warning" role="alert">
+          <div className="spinner-border alertSpinner" role="status">
+            <span className="sr-only">Loading...</span>
+          </div>
+          <div className="alertText">
+            Verifying your credentials (should be a few minutes). You'll be able
+            to edit/add once verified.
+          </div>
+        </div>
+      </Box>
+    );
+  }
+
+  // TODO: check if this accounts for mismatched validation, ie email does not match facility
+  if (!verified && badDomain) {
+    content = (
+      <Box>
+        <div className="alert alert-danger" role="alert">
+          <div className="alertText">
+            Your email doesn't look like it's from a healthcare provider. Please{' '}
+            <a
+              href={Routes.LOGOUT}
+              style={{
+                color: '#721c24',
+                fontWeight: 'bold',
+                textDecoration: 'underline',
+              }}
+            >
+              log out
+            </a>{' '}
+            and try your work email or contact{' '}
```

Comment:
We should have `locales` for these strings.

Reply:
Actually we have an issue for converting these alerts to a shared component. We can handle the strings with that one so never mind!